ext/ftp/ftp.h

```diff
@@ -39,6 +39,11 @@
 /* XXX this should be configurable at runtime XXX */
 #define	FTP_BUFSIZE	4096
 
+#define FTP_MLSD_LINE_LENGTH_MAX 4096
+#define FTP_MLSD_CHUNK_LENGTH_MAX 256
```

Comment:
The [RFC:3659 Extensions to FTP](https://tools.ietf.org/html/rfc3659) don't specify any limits for the length of lines, fact name, fact value or the file name. It just seemed reasonable limits for me.